Workaround for static builds
--- a/build/meson/meson.build
+++ b/build/meson/meson.build
@@ -96,11 +96,11 @@ else
   use_multi_thread = thread_dep.found()
 endif
 # Arguments in dependency should be equivalent to those passed to pkg-config
-zlib_dep = dependency('zlib', required: feature_zlib)
+zlib_dep = dependency('zlib', required: feature_zlib, static: true)
 use_zlib = zlib_dep.found()
-lzma_dep = dependency('liblzma', required: feature_lzma)
+lzma_dep = dependency('liblzma', required: feature_lzma, static: true)
 use_lzma = lzma_dep.found()
-lz4_dep = dependency('liblz4', required: feature_lz4)
+lz4_dep = dependency('liblz4', required: feature_lz4, static: true)
 use_lz4 = lz4_dep.found()
 
 # =============================================================================
