Workaround for static builds
--- a/meson.build	2025-06-10 19:18:29.459461798 +0000
+++ b/meson.build2	2025-06-10 19:18:24.451461908 +0000
@@ -9,7 +9,7 @@ project('pax-utils', 'c',
 
 cc = meson.get_compiler('c')
 
-libcap = dependency('libcap', required : get_option('use_libcap'))
+libcap = dependency('libcap', required : get_option('use_libcap'), static: true)
 if libcap.found()
   add_project_arguments('-DWANT_SYSCAP', language : 'c')
 endif
